pkg/controllers/csi/metadata/models.go

```diff
@@ -0,0 +1,64 @@
+package metadata
+
+import (
+	"time"
+
+	"gorm.io/gorm"
+)
+
+type TimeStampedModel struct {
+	CreatedAt time.Time      `json:"createdAt"`
+	UpdatedAt time.Time      `json:"updatedAt"`
+	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
+}
+
+// TenantConfig holds info about a given configuration for a tenant.
+type TenantConfig struct {
+	CodeModule CodeModule `gorm:"foreignKey:Version;references:DowloadedCodeModuleVersion"`
```

Comment:
Would be very cool if you could make the diagram we talked about on the daily, because it's not clear to me in this case how the table would look like

Reply:
done.

Reply:
Thank you!